apps/api/src/app/shared/services/distributed-lock/distributed-lock.service.ts

```diff
@@ -0,0 +1,149 @@
+import Redis from 'ioredis';
+import * as Redlock from 'redlock';
+import { setTimeout } from 'timers/promises';
+import { Logger } from '@nestjs/common';
+
+const LOG_CONTEXT = 'DistributedLock';
+
+const redisUrl =
+  process.env.REDIS_HOST && process.env.REDIS_PORT
+    ? `${process.env.REDIS_HOST}:${process.env.REDIS_PORT}`
+    : 'localhost:6379';
```

Comment:
I actually think we can throw an error here instead of falling back to localhost, wdyt? We also have a validation in the env-validator for this value as a required value. wdyt?

Reply:
Let's go for it, to have extra safety in case environment validator somehow stops working.